connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -133,6 +133,7 @@ class SparkSession(private val client: SparkConnectClient, private val cleaner:
 
   override def close(): Unit = {
     client.shutdown()
+    cleaner.stop()
```

Comment:
Like this? stop cleaner and clean up close ref in `referenceBuffer` that has not been cleaned? @zhenlineo

Reply:
I am looking at this afternoon. We cannot directly shutdown the cleaner as it is shared by all sessions. I have some draft [here](https://github.com/zhenlineo/spark/pull/new/allocator-memleak). Be free to use any idea there. If you do not have more time on this. I will add some testing and send a PR tomorrow.